compute_sdk/globus_compute_sdk/sdk/client.py

```diff
@@ -387,24 +373,29 @@ def run(self, *args, endpoint_id=None, function_id=None, **kwargs) -> str:
         assert endpoint_id is not None, "endpoint_id key-word argument must be set"
         assert function_id is not None, "function_id key-word argument must be set"
 
-        batch = self.create_batch(create_websocket_queue=self.asynchronous)
-        batch.add(function_id, endpoint_id, args, kwargs)
+        batch = self.create_batch(endpoint_id, create_websocket_queue=self.asynchronous)
+        batch.add(function_id, args, kwargs)
         r = self.batch_run(batch)
 
-        return r[0]
+        return r["tasks"][function_id][0]
 
-    def create_batch(self, task_group_id=None, create_websocket_queue=False) -> Batch:
+    def create_batch(
+        self, endpoint_id, task_group_id=None, create_websocket_queue=False
+    ) -> Batch:
```

Comment:
While we're here, let's update the signature with typing information.

(And props if using 3.10+ syntax, like `: uuid.UUID | str`.)